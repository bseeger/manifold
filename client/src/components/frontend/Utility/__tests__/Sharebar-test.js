import React from 'react';
import renderer from 'react-test-renderer';
import build from 'test/fixtures/build';
import entity from 'test/fixtures/entity';
import { Provider } from 'react-redux';
import Sharebar from '../Sharebar';

describe("Frontend.Utility.Sharebar component", () => {

  const store = build.store();
  const settings = entity.defaults.settings;

  it('renders correctly', () => {
    const component = renderer.create(
      <Provider store={store} >
        <Sharebar
          dispatch={store.dispatch}
          label="The Label"
          url="the/url"
          message="The Message"
          settings={settings}
        />
      </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
