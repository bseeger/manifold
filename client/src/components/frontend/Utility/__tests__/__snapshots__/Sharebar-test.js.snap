exports[`Frontend.Utility.Sharebar component renders correctly 1`] = `
<nav
  className="share-nav-primary">
  <span>
    The Label
  </span>
  <ul>
    <li>
      <button
        onClick={[Function]}
        target="_blank">
        <i
          className="manicon manicon-twitter" />
      </button>
    </li>
  </ul>
</nav>
`;
